packages/components/src/presentation-components/properties/editors/NavigationPropertyEditorContext.tsx

```diff
@@ -0,0 +1,78 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+/**
+ * @packageDocumentation
+ * @module Properties
+ */
+
+import { createContext, PropsWithChildren, useContext, useMemo } from "react";
+import { PropertyDescription } from "@itwin/appui-abstract";
+import { IModelConnection } from "@itwin/core-frontend";
+import { NavigationPropertyInfo } from "@itwin/presentation-common";
+import { IContentDataProvider } from "../../common/ContentDataProvider";
+
+/**
+ * A context providing necessary information for [[NavigationPropertyEditor]].
+ * @public
+ */
+export interface NavigationPropertyEditorContext {
+  /** iModel connection to pull data from. */
+  imodel: IModelConnection;
+  /** Getter to get [NavigationPropertyInfo]($presentation-common) for specific property. */
+  getNavigationPropertyInfo: (property: PropertyDescription) => Promise<NavigationPropertyInfo | undefined>;
+}
+
+const navigationPropertyEditorContext = createContext<NavigationPropertyEditorContext | undefined>(undefined);
+
+/**
+ * Props for [[NavigationPropertyEditorContextProvider]].
+ * @public
+ */
+export interface NavigationPropertyEditorContextProviderProps {
+  /** iModel connection to pull data from. */
+  imodel: IModelConnection;
+  /** Getter to get [NavigationPropertyInfo]($presentation-common) for specific property. */
+  getNavigationPropertyInfo: (property: PropertyDescription) => Promise<NavigationPropertyInfo | undefined>;
+}
+
+/**
+ * Creates context that supplies necessary navigation property-related information for [[NavigationPropertyEditor]].
+ * @public
+ */
+export function NavigationPropertyEditorContextProvider({ children, ...props }: PropsWithChildren<NavigationPropertyEditorContextProviderProps>) {
+  return <navigationPropertyEditorContext.Provider value={props}>{children}</navigationPropertyEditorContext.Provider>;
+}
+
+/**
+ * Returns context provided by [[NavigationPropertyEditorContextProvider]]. The context is required for rendering [[NavigationPropertyEditor]] with editable
+ * navigation property values.
+ * @public
+ */
+export function useNavigationPropertyEditorContext() {
```

Comment:
I don't think this needs to be exported. It should not be used outside NavigationPropertyEditor.

Reply:
made `NavigationPropertyEditorContext` and `useNavigationPropertyEditorContext` internal